### ISSUE TITLE
💡 RFC: ChordSheetParser to parse Sections headings as comments

### ISSUE THREAD

**Issue by isaiahdahl** [PraiseCharts/ChordSheetJS#217]
Parent Task: https://github.com/PraiseCharts/studio/issues/32
### Background & Motivation

I want this:
```
Chorus 1
                     Dsus  
There’s honey in the rock, water in the stone

Verse 1
D        Dsus               D        Dsus 
  Praying     for a miracle,  thirsty
```

To generate this chordpro
```
{c:Chorus 1}
There’s honey in the [Dsus]rock, water in the stone

{c:Verse 1}
[D] Praying[Dsus] for a miracle,[D] thirsty[Dsus]
```

![CleanShot 2022-07-21 at 14 20 03](https://user-images.githubusercontent.com/23046374/180317298-738b2762-6308-464f-bbe6-044af3f77512.gif)

### Possible solutions

- first line in a ChordSheet will be a comment.
- if there's a line break the following line will be a comment

**Comment by martijnversluis** [comment]
> first line in a ChordSheet will be a comment.

This could go wrong for something like:

    Intro: | A | D | F#m | E | (2x)

> if there's a line break the following line will be a comment

That won't work when separating sections with a newline:

```
How I [C]long to [Csus4]breathe the air of h[C]eaven
Where pain is [Am]gone, and [F]mercy fills the [G]streets

To look u[C]pon the [F]one who bled to s[C]ave m[G/B]e[Am]
And walk with [F]him for [G]all etern[C]ity[Csus4][C]
``` 

What about:

_A single line of text, that does not include any chord, is considered a comment_

**Comment by isaiahdahl** [comment]
My initial thoughts to your thoughts would be that the first problem could be solved by just considering that as an invalid way to write it out. Encouraging and documenting that it should be written like this. Syntax Highlighting & real time rendering in the editor will also help with this.

```
Intro (2x)
| A | D | F#m | E |
```

For the second issue:
- It could be encouraged that sections should not be broken apart with a new line. 
- Or possibly it could allow for the line break because the following line has chords in it.

My only problem with your solution
> A single line of text, that does not include any chord, is considered a comment

Is that we commonly come across lyrics-only lines in our chord charts. 
Here's a very simple example:
![CleanShot 2022-08-15 at 10 34 21@2x](https://user-images.githubusercontent.com/23046374/184686071-295a35fa-b96c-4b50-973d-d5ee3c2b1dab.png)

Another Idea would be to create a new `ChordsOverWordsParser` with a more strict standard for writing "Chords Over Words" style chord charts. Where the goal is to make it so the ChordPro & ChordsOverWords parsers/formatters are seamlessly compatible with each other.

- Metadata could be written as "Key: Value" (possible only certain metadata's supported)
- Maybe consider OnSong style first line is title, second line is artist
- Sections broken up by new lines
- Line that follows a line break is a comment.
- Possibly auto-apply  {start_of_x}...{end_of_x} sections based on the comment, where we'd look for special key words (Chorus Verse, V., Intro)

**Comment by isaiahdahl** [comment]
This is sorta a duplicate, or at least very related to #219

**Comment by isaiahdahl** [comment]
Closed by #272
